cmake/AddHostToBuildInfoFile.cmake

```diff
@@ -0,0 +1,10 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+file(APPEND
```

Comment:
Perhaps name this file SetSiteName or something like that and include the `site_name` command, or the other way round: inline the `file` command in the root CMakeLists.txt and delete this file?